tools/certgen/certgen.go

```diff
@@ -27,13 +27,27 @@ import (
 )
 
 var (
+	// serviceAddresses is a list of all the HTTP hostname:port combinations for generating a TLS certificate.
+	// It appears that gRPC does not care about validating the port number so we only add the HTTP addresses here.
+	serviceAddressList = []string{
```

Comment:
I'm working on a change to add the hostname to the config file. Do you mind adding a TODO at here so that we could generate the certificates based on values from the config in the future.

Reply:
I think the more sane approach is to specify the flag value if you're going to do auto generation.